vignettes/tidyselect.Rmd

```diff
@@ -0,0 +1,310 @@
+vignette](https://tidyselect.r-lib.org/articles/syntax.html). In this
+vignette, we describe how to include tidyselect in your own packages.
+
+
+## Before we start
+
+### Selections as dots or as named arguments
+
+There are two major ways of designing a function that takes
+selections.
+
+- __Variadic dots__ as in `dplyr::select()`.
+
+    ```{r, eval = FALSE}
+    mtcars %>% dplyr::select(mpg, cyl)
+    ```
+
+- __Named arguments__ as in `tidyr::pivot_longer()`. In this case,
+  multiple inputs can be provided inside `c()` or by using boolean
+  operators:
+
+    ```{r, eval = FALSE}
+    mtcars %>% pivot_longer(c(mpg, cyl))
+    mtcars %>% pivot_longer(mpg | cyl)
+    ```
+
+Our general recommendation is to take dots when the main purpose of
+the function is to create a new data structure based on a selection.
+When the selection is accessory to the main purpose of the function,
+take it as a named argument. For more advice about this, see the
+[Making data with `...`](https://design.tidyverse.org/dots-data.html)
+section of the tidyverse design book.
+
+
+### Do you need tidyselect?
+
+The tools described in this vignette are rather low level. Depending
+on your use case, it may be easier to wrap around `dplyr::select()`.
+You'll get a data frame containing the columns selected by your user,
+which you can then handle in various ways.
+
+The following examples illustrate how you could write a function that
+takes a selection of data and returns the corresponding data frame
+with capitalised names:
+
+```{r, eval = FALSE}
+# Passing dots
+toupper_dots <- function(data, ...) {
+  sel <- dplyr::select(data, ...)
+  rlang::set_names(sel, toupper)
+}
+# Interpolating a named argument with {{ }}
+toupper_arg <- function(data, arg) {
+  sel <- dplyr::select(data, {{ arg }})
+  rlang::set_names(sel, toupper)
+}
+
+mtcars %>% toupper_dots(mpg:disp, vs)
+#> # A tibble: 32 x 4
+#>     MPG   CYL  DISP    VS
+#>   <dbl> <dbl> <dbl> <dbl>
+#> 1  21       6   160     0
+#> 2  21       6   160     0
+#> 3  22.8     4   108     1
+#> 4  21.4     6   258     1
+#> # … with 28 more rows
+
+mtcars %>% toupper_arg(c(mpg:disp, vs))
+#> # A tibble: 32 x 4
+#>     MPG   CYL  DISP    VS
+#>   <dbl> <dbl> <dbl> <dbl>
+#> 1  21       6   160     0
+#> 2  21       6   160     0
+#> 3  22.8     4   108     1
+#> 4  21.4     6   258     1
+#> # … with 28 more rows
+```
+
+The main advantage of the lower level tidyselect tools is that they
+offer a bit more information and flexibility. Instead of returning the
+selected data, they return the __positions__ of selected elements
+inside the input data. If you don't need the selected positions and
+can afford the dependency, you may consider wrapping around dplyr
+instead.
+
+
+## The selection evaluators
+
+tidyselect is implemented with non-standard evaluation (NSE). This
+unique feature of the R language refers to the ability of functions to
+__defuse__ some or all of their arguments, and resume evaluation later
```

Comment:
Maybe use "delay execution" as well?

Reply:
I thought we were going for "defuse", from the booby trap analogy?

Reply:
Sorry, I meant additionally as a parenthetical since not everyone will know about the new terminology

Reply:
Now mentions "delay execution", and has a footnote comparing to "quoting".